src/include/parser/expression/column_value_expression.h

```diff
@@ -0,0 +1,209 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+#include "catalog/catalog_defs.h"
+#include "parser/expression/abstract_expression.h"
+
+namespace terrier::parser {
+
+/**
+ * Represents a column tuple value.
+ */
+class ColumnValueExpression : public AbstractExpression {
+ public:
+  /**
+   * This constructor is called only in postgresparser, setting the column name,
+   * and optionally setting the table name and alias.
+   * Namespace name is always set to empty string, as the postgresparser does not know the namespace name.
+   * Parameter namespace name is included so that the program can differentiate this constructor from
+   * another constructor that sets the namespace name, table name. and column name.
+   * @param namespace_name namespace name
+   * @param table_name table name
+   * @param col_name column name
+   * @param alias alias of the expression
+   */
+  ColumnValueExpression(std::string namespace_name, std::string table_name, std::string col_name, std::string alias)
+      : AbstractExpression(ExpressionType::COLUMN_VALUE, type::TypeId::INVALID, std::move(alias), {}),
+        namespace_name_(std::move(namespace_name)),
+        table_name_(std::move(table_name)),
+        column_name_(std::move(col_name)) {}
+
+  /**
+   * @param table_name table name
+   * @param col_name column name
+   */
+  ColumnValueExpression(std::string table_name, std::string col_name)
+      : AbstractExpression(ExpressionType::COLUMN_VALUE, type::TypeId::INVALID, {}),
+        table_name_(std::move(table_name)),
+        column_name_(std::move(col_name)) {}
+
+  /**
+   * @param namespace_name namespace name
+   * @param table_name table name
+   * @param col_name column name
+   */
+  ColumnValueExpression(std::string namespace_name, std::string table_name, std::string col_name)
+      : AbstractExpression(ExpressionType::COLUMN_VALUE, type::TypeId::INVALID, {}),
+        namespace_name_(std::move(namespace_name)),
+        table_name_(std::move(table_name)),
+        column_name_(std::move(col_name)) {}
+
+  /**
+   * @param database_oid database OID
+   * @param table_oid table OID
+   * @param column_oid column OID
+   */
+  ColumnValueExpression(catalog::db_oid_t database_oid, catalog::table_oid_t table_oid, catalog::col_oid_t column_oid)
```

Comment:
Missing `namespace_oid_t`

Reply:
We don’t need it here because we have the table OID.  Except for creating a table, we do not need the namespace if we already have the object OID.

Reply:
Also I'm not clear when is this constructor called - certainly not in the parser, as the parser has no way to know oids.

Reply:
The binder should replace most (all?) of the ColumValueExpressions created by the parser with ones made through this constructor.